app/assets/javascripts/save-on-change.coffee

```diff
@@ -179,21 +179,29 @@ class @ForwardBlocker
     @binding.bind 'navigate_away', (e,opts) =>
       @navigate_away(opts.click_element, opts.action_to_perform)
 
-  prevent_forward_navigation_for: (blocker) ->
-    if blocker not in @blockers
-      @blockers.push blocker
+  get_blocker_index: (blocker) ->
+    firstBlocker = (b for b in @blockers when b.blocker is blocker)[0]
+    @blockers.indexOf(firstBlocker)
```

Comment:
This is confusing. There is an array called "blockers", but its values are actually objects that have a field named `blocker`, and a field named `disabledForwardNavMessage`.   I think calling the array `blockers` makes sense. I think the `blocker` field is actually the dom element, so I'd call that `element`.